### ISSUE TITLE
`check_model` check notmality in GLMs?

### ISSUE THREAD

**Issue by mattansb** [easystats/performance#501]
Is there a reason for this? What am I missing...

``` r
library(performance)

set.seed(3)
mu <- rpois(500, lambda = 3)
x <- rnorm(500, mu, mu*3) |> ceiling() |> pmax(0)

quine.nb1 <- MASS::glm.nb(x ~ mu)

check_normality(quine.nb1)
#> Checking normality of residuals is only useful an appropriate assumption
#>   for linear models.
#> NULL
```

But…

``` r
check_model(quine.nb1)
```

![](https://i.imgur.com/fY2qVYP.png)

<sup>Created on 2022-10-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by bwiernik** [comment]
the current normality check isn't really appropriate generally except for linear models. something like the DhARMA QQ plots would be more appropriate

**Comment by strengejacke** [comment]
_MASS_ package provides a QQ plot (with `plot()`), using std. Pearson resid. Maybe we fall back to this plot for now?

**Comment by strengejacke** [comment]
Wait, we're already doing this:

``` r
library(performance)
set.seed(3)
mu <- rpois(500, lambda = 3)
x <- rnorm(500, mu, mu*3) |> ceiling() |> pmax(0)

quine.nb1 <- MASS::glm.nb(x ~ mu)
check_model(quine.nb1, check = "qq")
```

![](https://i.imgur.com/mnFNWfK.png)

``` r
plot(quine.nb1)
```

![](https://i.imgur.com/n42GAdt.png)

**Comment by mattansb** [comment]
> MASS package provides a QQ plot (with plot()), using std. Pearson resid. Maybe we fall back to this plot for now?

Aren't these also irrelevant?

**Comment by bbolker** [comment]
This is a classic GLM approach. When sample sizes are reasonably large (e.g. Poisson with 'large' mean (>5?), Binomial with `n*p*(1-p)` 'large'), the Pearson residuals for a GLM approach normality. This is how Q-Q plotting was traditionally done before randomized/simulated residuals (don't know if this was done prior to Augustin et al 2012?) were computationally feasible (and hence why `plot.lm` is used for GLMs (whose class is `c("glm", "lm")`).  (Furthermore, when you plot `quine.nb1` above, the `plot()` method is *not coming from `MASS`*; instead, it's falling back to `stats::plot.lm` (`class(quine.nb1)` is `c("negbin", "glm", "lm")`.)


Augustin, Nicole H., Erik-André Sauleau, and Simon N. Wood. 2012. “On Quantile Quantile Plots for Generalized Linear Models.” Computational Statistics & Data Analysis 56 (8): 2404–9. https://doi.org/10.1016/j.csda.2012.01.026.

**Comment by strengejacke** [comment]
Also note that we have changed the behaviour and adopted the new computation for QQ-plot for GLMs that is implemented in R 4.3 (https://github.com/easystats/performance/issues/571). That maybe resolves this issue?

**Comment by bbolker** [comment]
To be honest I don't fully understand the advantages of the base-R change, but I'm fine with this. I think it's a matter of taste whether you want to present these plots, or take something from `mgcv::qq.gam` or `DHARMa`, or not show anything at all ...

**Comment by epertham** [comment]
> the current normality check isn't really appropriate generally except for linear models. something like the DhARMA QQ plots would be more appropriate

Hello, thanks a lot for this package very useful and user-frienfly. What do you think about integrating the DHARMA check of residuals for (generalized) mixed models ? Best, Emeline

**Comment by strengejacke** [comment]
We have an older issue where we discussed integrating DHARMa (#376), but I personally have rarely used that package, and I'm not sure if other easystats-members currently find the time to work on this. But it's definitely not forgotten...

**Comment by strengejacke** [comment]
Should be fixed in #643

``` r
library(performance)
set.seed(3)
mu <- rpois(500, lambda = 3)
x <- rnorm(500, mu, mu * 3) |>
  ceiling() |>
  pmax(0)

quine.nb1 <- MASS::glm.nb(x ~ mu)
check_normality(quine.nb1)
#> There's no formal statistical test for normality for generalized linear
#>   model.
#>   Instead, please use `simulate_residuals()` and `check_residuals()` to
#>   check for uniformity of residuals.
check_residuals(quine.nb1)
#> Warning: Non-uniformity of simulated residuals detected (p < .001).
plot(simulate_residuals(quine.nb1))
```

![](https://i.imgur.com/0Lo79BQ.png)

<sup>Created on 2024-03-15 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>
